<launch>
    <!-- Load the gain matrix from config file -->
    <rosparam command="load" file="$(find sys_iden)/config/iden_config.yaml"/>
    
    <node name="serial_node"        pkg="rosserial_python"      type="serial_node.py" output="screen">
        <param name="port"              type="string"               value="/dev/ttyACM0"/>
        <param name="baud"              type="int"                  value="57600"/>
    </node>
    <!-- Delay to stabilize low level -->
    <arg name="node_start_delay" default="15.0" /> 
    
    <!-- Declare the parameters for excitation state and gain matrix -->
    <arg name = "excitation_state"/>

    <!-- Launch the sys_idn_node.py node with specified parameters -->
    <node name="thrust_controller_node" pkg="sys_iden" type="thrust_controller_node.py" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node name="sys_idn_node" pkg="sys_iden" type="sys_idn_node.py" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
        <param name="excitation_state" value="$(arg excitation_state)"/>
    </node>
</launch>
